src/secureCreatives.js

```diff
@@ -93,23 +93,25 @@ function resizeRemoteCreative({ adUnitCode, width, height }) {
   });
 
   function getElementByAdUnit(elmType) {
-    let id = getElementIdBasedOnAdServer(adUnitCode);
+    let id = getElementIdBasedOnAdServer(adId, adUnitCode);
     let parentDivEle = document.getElementById(id);
     return parentDivEle && parentDivEle.querySelector(elmType);
   }
 
-  function getElementIdBasedOnAdServer(adUnitCode) {
+  function getElementIdBasedOnAdServer(adId, adUnitCode) {
     if (window.googletag) {
-      return getDfpElementId(adUnitCode)
+      return getDfpElementId(adId)
     } else if (window.apntag) {
       return getAstElementId(adUnitCode)
     } else {
       return adUnitCode;
     }
   }
 
-  function getDfpElementId(adUnitCode) {
-    return find(window.googletag.pubads().getSlots().filter(isSlotMatchingAdUnitCode(adUnitCode)), slot => slot).getSlotElementId()
+  function getDfpElementId(adId) {
+    return find(window.googletag.pubads().getSlots().filter(s => {
+      return s.getTargetingMap()['hb_adid'].includes(adId);
```

Comment:
Please use `includes` from core-js https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L10

Reply:
> I have 2 questions here.
> 
> 1. standard adserverTargeting can be disabled if you use `pbjs.bidderSettings.adserverTargeting.standard` to configure your own. So someone may not send hb_adId and use something else. Let me know if this is invalid use case.
> 2. `getTargetingMap` function is not documented here https://developers.google.com/doubleclick-gpt/reference#googletag.slot. But I tested on page and it exists so not sure if we should use undocumented function.

Good one, we could just use the getTargeting(key) instead.